src/Servers/HttpSys/test/FunctionalTests/Http2Tests.cs

```diff
@@ -0,0 +1,138 @@
+
+                    h2Connection.Logger.LogInformation("Received headers in a single frame.");
+
+                    h2Connection.ResetHeaders();
+                    decodedHeaders = h2Connection.DecodeHeaders(headersFrame);
+
+                    // HTTP/2 filters out the connection header
+                    Assert.False(decodedHeaders.ContainsKey(HeaderNames.Connection));
+                    Assert.Equal("200", decodedHeaders[HeaderNames.Status]);
+
+                    await h2Connection.StopConnectionAsync(expectedLastStreamId: 1, ignoreNonGoAwayFrames: false);
+
+                    h2Connection.Logger.LogInformation("Connection stopped.");
+                })
+                .Build().RunAsync();
+        }
+
+        [ConditionalFact]
+        [MinimumOSVersion(OperatingSystems.Windows, WindowsVersions.Win10_19H2, SkipReason = "GoAway support was added in Win10_19H2.")]
+        public async Task ConnectionClose_OSSupport_SendsGoAway()
+        {
+            using var server = Utilities.CreateDynamicHttpsServer(out var address, httpContext =>
+            {
+                httpContext.Response.Headers[HeaderNames.Connection] = "close";
+                return Task.FromResult(0);
+            });
+
+            await new HostBuilder()
+                .UseHttp2Cat(address, async h2Connection =>
+                {
+                    await h2Connection.InitializeConnectionAsync();
+
+                    h2Connection.Logger.LogInformation("Initialized http2 connection. Starting stream 1.");
+
+                    await h2Connection.StartStreamAsync(1, Http2Utilities.BrowserRequestHeaders, endStream: true);
+
+                    var goAwayFrame = await h2Connection.ReceiveFrameAsync();
+                    h2Connection.VerifyGoAway(goAwayFrame, int.MaxValue, Http2ErrorCode.NO_ERROR);
+
+                    var headersFrame = await h2Connection.ReceiveFrameAsync();
+
+                    Assert.Equal(Http2FrameType.HEADERS, headersFrame.Type);
+                    Assert.Equal(Http2HeadersFrameFlags.END_HEADERS, headersFrame.HeadersFlags);
+
+                    h2Connection.Logger.LogInformation("Received headers in a single frame.");
+
+                    var decodedHeaders = h2Connection.DecodeHeaders(headersFrame);
+
+                    // HTTP/2 filters out the connection header
+                    Assert.False(decodedHeaders.ContainsKey(HeaderNames.Connection));
+                    Assert.Equal("200", decodedHeaders[HeaderNames.Status]);
+
+                    var dataFrame = await h2Connection.ReceiveFrameAsync();
+                    Assert.Equal(Http2FrameType.DATA, dataFrame.Type);
+                    Assert.Equal(Http2DataFrameFlags.END_STREAM, dataFrame.DataFlags);
+                    Assert.Equal(0, dataFrame.PayloadLength);
+
+                    // TODO: Why doesn't HttpSys send a final GoAway or close the connection?
```

Comment:
So is there a goaway sent at all right now?

Reply:
I followed up with them on e-mail and I'll update the test. Http.Sys sends the initial GoAway, but considers the final GoAway redundant since the client already sent one. They will only send a final GoAway if the client sends 100 streams after the initial GoAway.